### PR TITLE
feat: support graphql-ws client for GraphiQL IDE

### PR DESCRIPTION
**Pull request by junminstorage**
To follow up #687 and continue to address #390, I drafted this PR to 
- [x] add support for https://github.com/enisdenjo/graphql-ws client for GraphiQL IDE.
- [x] upgrade subscription-transport-ws to its latest version
- [x] add example server startup script of using graphql-ws for subscription support
- [x] add example server startup script of using subscriptions-transport-ws for subscription support
- [x] use fetcher from https://github.com/graphql/graphiql

Developer provides ```websocketClient``` as part of ```graphiql``` option, its values are "v0" for subscriptions-transport-ws, "v1" for graphql-ws, I did this based on graphile/postgraphile#1338, open to better naming suggestion. 

Specifically check out this tutorial script about how to start web socket server to support subscription using enisdenjo/graphql-ws : https://github.com/junminstorage/graphql-tutorial/blob/master/src/index_v1.js 

You can also checkout the two newly added examples inside /examples directory in this PR for usage.

### CODE REVIEW

**Comment by junminstorage** [comment]
@acao this is the newly created PR. it looks all recently opened PRs failed on tests running on Node v12. Do you know why?

**Comment by acao** [comment]
@junminstorage not sure, appears to be the same issue on a few PRs and I can’t seem to find the actual error! Looking into it tonight, we will have this and all the subscriptions support docs released by friday at latest

**Comment by junminstorage** [comment]
@acao I think it is related to this https://github.com/nodejs/undici/issues/702

File: package.json [hunk 1]
```diff
@@ -83,9 +87,10 @@
     "eslint-plugin-istanbul": "0.1.2",
     "eslint-plugin-node": "11.1.0",
     "express": "4.17.1",
-    "graphiql": "1.0.6",
+    "graphiql": "^1.4.1",
```
Comment (justinmchase) [inline package.json:90]: 👍 

Currently you can't provide request headers when using graphiql, getting this updated version in here would be very helpful.

**Review by justinmchase — APPROVED** [review]

**Comment by justinmchase** [comment]
All of the failing tests for node 12 have the errors `DeprecationWarning: The legacy HTTP parser is deprecated.`

**Comment by junminstorage** [comment]
right, it is documented here: https://github.com/graphql/express-graphql/issues/756. 
@acao is it possible to continue to merge this PR and release it.

**Comment by justinmchase** [comment]
I agree it's probably something that can be fixed later. It appears that it would not create a regression.

**Comment by acao** [comment]
@junminstorage let me know if there’s anything I can help with here! Do you think you’ll need any more patches to GraphiQL?

**Comment by acao** [comment]
@junminstorage ok we're good to merge here it looks like! thanks everyone for the awesome work!

**Comment by junminstorage** [comment]
@acao I fixed the mocha issue as stated in #762 , this PR is ready to go.

**Comment by acao** [comment]
to note for everyone here! 

this PR must be merged for `@stream` and `@defer` support to become stable:
https://github.com/graphql/graphql-js/pull/2839

**Pull request merged**